app/scripts/modules/core/src/application/nav/ApplicationNavigation.spec.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { BehaviorSubject } from 'rxjs';
+import { mock } from 'angular';
+import { UIRouterReact, UIRouterContext } from '@uirouter/react';
+import { StateMatcher } from '@uirouter/core';
+
+import { ReactInjector, REACT_MODULE } from '../../reactShims';
+import { OVERRIDE_REGISTRY } from '../../overrideRegistry';
+import {
+  mockAppConfigDataSourceConfig,
+  mockServerGroupDataSourceConfig,
+  mockLoadBalancerDataSourceConfig,
+  mockTaskDataSourceConfig,
+  mockPipelineDataSourceConfig,
+} from '@spinnaker/mocks';
+import { ApplicationModelBuilder } from '../../application';
+import { IPipeline } from '../../domain';
+import { ApplicationDataSource } from '../service/applicationDataSource';
+
+import { ApplicationNavigation } from './ApplicationNavigation';
+
+describe('ApplicationNavigation', () => {
+  let $uiRouter: UIRouterReact;
+  const currentStates = ['**.pipelines.**', '**.tasks.**'];
+
+  beforeEach(mock.module(REACT_MODULE, OVERRIDE_REGISTRY));
+  beforeEach(
+    mock.inject((_$uiRouter_: UIRouterReact) => {
+      $uiRouter = _$uiRouter_;
+      // Inititalize base route
+      $uiRouter.stateRegistry.root().name = 'http://localhost:9000';
+    }),
+  );
+  beforeEach(() => {
+    // Initialize current route
+    spyOn(ReactInjector.$state, 'includes').and.callFake((substate: any) => currentStates.includes(substate));
+    spyOn(StateMatcher.prototype, 'resolvePath').and.callFake(() => 'pipelines.executions');
```

Comment:
this is an extremely internal API that shouldn't be relied upon.  I'll see if I can come up with a better way to handle this.

Reply:
Sounds good. Maybe moving to `useCurrentStateandParams` hook will help. I had to stub anything relating to the `ReactInjector` for mocking an active category or category change.

Reply:
I guess I was wrong.  After looking into the docs, this is indeed marked as public API.
However, this might be a slightly more idiomatic way to mock this:
```suggestion
    spyOn(StateMatcher.prototype, 'find').and.callFake(() => undefined as any);
```